### PR TITLE
CLN 31942/replace appender with doc 3

### PR DESCRIPTION
**Pull request by smartvinnetou**
- [x] ref #31942
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] replaces `Appender` decorator with `doc`

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -2102,22 +2102,24 @@ def to_feather(self, path, **kwargs) -> None:
...
-        ... )
-        >>> print(df.to_markdown())
-        |    | animal_1   | animal_2   |
-        |---:|:-----------|:-----------|
-        |  0 | elk        | dog        |
-        |  1 | pig        | quetzal    |
-        """
+    @doc(
+        Series.to_markdown,
+        klass=_shared_doc_kwargs["klass"],
```
Comment (WillAyd) [inline pandas/core/frame.py:2107]: Shouldn’t klass here be DataFrame?

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py (hunk 1 above)
Comment (smartvinnetou) [reply pandas/core/frame.py:2107]: `_shared_doc_kwargs["klass"]=="DataFrame"` Using the dictionary is more consistent with other places in this file where the decorator is used.

Comment (smartvinnetou) [reply pandas/core/frame.py:2107]: @WillAyd do you accept my reasoning?

**Comment by smartvinnetou** [comment]
After a rebase with the upstream master, there is a failing test because of a missing S3 bucket:

`def test_write_s3_csv_fails(self, tips_df):
        # GH 32486
        # Attempting to write to an invalid S3 path should raise
        with pytest.raises(
            FileNotFoundError, match="The specified bucket does not exist"
        ):
>           tips_df.to_csv("s3://an_s3_bucket_data_doesnt_exit/not_real.csv")
pandas/tests/io/parser/test_network.py:178: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pandas/core/generic.py:3171: in to_csv
    formatter.save()
pandas/io/formats/csvs.py:227: in save
    f.close()
../../../miniconda3/envs/pandas-dev/lib/python3.7/site-packages/fsspec/spec.py:1313: in close
    self.flush(force=True)
../../../miniconda3/envs/pandas-dev/lib/python3.7/site-packages/fsspec/spec.py:1187: in flush
    if self._upload_chunk(final=force) is not False:
../../../miniconda3/envs/pandas-dev/lib/python3.7/site-packages/s3fs/core.py:1252: in _upload_chunk
    self.commit()
../../../miniconda3/envs/pandas-dev/lib/python3.7/site-packages/s3fs/core.py:1269: in commit
    Key=self.key, Bucket=self.bucket, Body=data, **self.kwargs
../../../miniconda3/envs/pandas-dev/lib/python3.7/site-packages/s3fs/core.py:1131: in _call_s3
    **kwargs)
../../../miniconda3/envs/pandas-dev/lib/python3.7/site-packages/s3fs/core.py:200: in _call_s3
    return method(**additional_kwargs)
../../../miniconda3/envs/pandas-dev/lib/python3.7/site-packages/botocore/client.py:316: in _api_call
    return self._make_api_call(operation_name, kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <botocore.client.S3 object at 0x7f4df82d5990>
operation_name = 'PutObject'
api_params = {'Body': <_io.BytesIO object at 0x7f4df82b9e90>, 'Bucket': 'an_s3_bucket_data_doesnt_exit', 'Key': 'not_real.csv'}
    def _make_api_call(self, operation_name, api_params):
        operation_model = self._service_model.operation_model(operation_name)
        service_name = self._service_model.service_name
        history_recorder.record('API_CALL', {
            'service': service_name,
            'operation': operation_name,
            'params': api_params,
        })
        if operation_model.deprecated:
            logger.debug('Warning: %s.%s() is deprecated',
                         service_name, operation_name)
        request_context = {
            'client_region': self.meta.region_name,
            'client_config': self.meta.config,
            'has_streaming_input': operation_model.has_streaming_input,
            'auth_type': operation_model.auth_type,
        }
        request_dict = self._convert_to_request_dict(
            api_params, operation_model, context=request_context)
    
        service_id = self._service_model.service_id.hyphenize()
        handler, event_response = self.meta.events.emit_until_response(
            'before-call.{service_id}.{operation_name}'.format(
                service_id=service_id,
                operation_name=operation_name),
            model=operation_model, params=request_dict,
            request_signer=self._request_signer, context=request_context)
    
        if event_response is not None:
            http, parsed_response = event_response
        else:
            http, parsed_response = self._make_request(
                operation_model, request_dict, request_context)
    
        self.meta.events.emit(
            'after-call.{service_id}.{operation_name}'.format(
                service_id=service_id,
                operation_name=operation_name),
            http_response=http, parsed=parsed_response,
            model=operation_model, context=request_context
        )
    
        if http.status_code >= 300:
            error_code = parsed_response.get("Error", {}).get("Code")
            error_class = self.exceptions.from_code(error_code)
>           raise error_class(parsed_response, operation_name)
E           botocore.errorfactory.NoSuchBucket: An error occurred (NoSuchBucket) when calling the PutObject operation: The specified bucket does not exist`

This looks unrelated to the changes in this PR.

**Comment by smartvinnetou** [comment]
Can someone help with merging this PR?

**Comment by WillAyd** [comment]
Looks like some merge conflicts in - can you fix that up?

**Comment by smartvinnetou** [comment]
> Looks like some merge conflicts in - can you fix that up?

done

**Review by WillAyd — APPROVED** [review]
@smartvinnetou for future reference would be nice if you can make smaller PRs. Something with this many changes is much harder to review at once, so typically better to have multiple PRs with smaller changes.

That said this lgtm - @datapythonista care to look?

**Comment by smartvinnetou** [comment]
> @smartvinnetou for future reference would be nice if you can make smaller PRs. Something with this many changes is much harder to review at once, so typically better to have multiple PRs with smaller changes.

@WillAyd I appreciate that. That's why I stopped making more changes. I started other branches but soon found that there was a non trivial overhead on maintaining them and keeping them current. I hit a pause button until PRs start to be merged.

**Comment by datapythonista** [comment]
@smartvinnetou can you please resolve the conflicts?

As said, smaller PRs are faster to review, get merged faster, and conflicts are less likely.

**Comment by smartvinnetou** [comment]
> As said, smaller PRs are faster to review, get merged faster, and conflicts are less likely.

@datapythonista, point taken. Will keep future PRs smaller.

**Comment by jreback** [comment]
@smartvinnetou pls merge master again.

@jorisvandenbossche @TomAugspurger @datapythonista can you give a quick look again and merge if good.

**Comment by smartvinnetou** [comment]
> @smartvinnetou pls merge master again.

Conflict resolved.

> 
> @jorisvandenbossche @TomAugspurger @datapythonista can you give a quick look again and merge if good.

**Comment by datapythonista** [comment]
Thanks @smartvinnetou, very nice clean up.

Once the development documentation is updated, can you please have a quick look at the updated pages, and see if everything is rendering as expected.

Thanks!

**Comment by smartvinnetou** [comment]
> Thanks @smartvinnetou, very nice clean up.
> 
> Once the development documentation is updated, can you please have a quick look at the updated pages, and see if everything is rendering as expected.
> 
> Thanks!

@datapythonista, where do I find this documentation?

**Comment by datapythonista** [comment]
https://pandas.pydata.org/docs/dev/

**Pull request merged**